### ISSUE TITLE
App hangs when memory reaches 2gb

### ISSUE THREAD

**Issue by sup3ruser** [johanneszab/TumblThree#67]
On v.42, app seems to hang when it reaches 2gb of memory.  I have 32gb of ddr4 3200mhz, so there is plenty of memory available for app to use if needed.  CPU is i7 12core with m.2 as primary ssd, hdd where scraping is done is on a 7200rpm, so system is pretty speedy & stable. 

Jerome

**Comment by johanneszab** [comment]
Try if [this gcAllowVeryLargeObjects configuration element](https://msdn.microsoft.com/en-us/library/hh285054(v=vs.110).aspx) helps by adding it to the TumblThree.exe.config file. It's just a guess since I most likely won't reproduce this as there is more important things to do and I won't download a crap ton of images just for this.

Please add the whole error message the next time and maybe let other people know if the option helps ..

**Comment by johanneszab** [comment]
If someone cares for this, patches are of course welcome if there is a _bug_.

**Comment by sup3ruser** [comment]
@johanneszab 
Im absolutely lost as how to actually add that code to the config file.  But here is an update that I came to state (not realizing that you already responded with a potential fix)

App (.42) hasn't been hanging (after I changed my scraping pattern) when approaching 2gb in memory.  I've actually seen it go as high as 4.5gb.  Only issue that I noticed then is that it becomes more laggy, less responsive.  

The bug "Crawling #70" is what caused me to change my scrape pattern, 1 blog at a time.  Then is when I stopped experiencing the hanging at 2gb of memory.  Also, I didn't get an error, because the app considered itself to be running fine.  Not only that, when it did hang, I just force closed it anyway.

Since the app gets less responsive when its using quite a bit of ram, Im guessing that it prob wont be responsive at all when you hit somewhere around the 10-20gb of memory mark, considering how unresponsive it becomes at only 4-5gb of memory.  Will try to figure out a clever way to get it use that much memory and report back whether or not this is even a bug for a different post.  But for now, I guess we can rule this out (cause other issues are more important, like crawling lol)
